guides/nx.md

```diff
@@ -0,0 +1,181 @@
+# NX
+
+[NX](https://nx.dev/) is a tool for managing monorepos.
+
+This guide shows you how to create a [connector](/guides/connectors) for your [NX](https://nx.dev/) application on the Moovweb XDN.
+
+In the example, we consider that default NX project uses [NextJS](https://nextjs.org/) framework
+
+## 1. Install XDN packages
+
+```sh
+yarn add -D @xdn/cli @xdn/core
+```
+
+## 2. Add XDN connector files
+
+<em>followed by [Writing a connector](https://developer.moovweb.com/guides/connectors) guide</em>
+
+```js
+// xdn/NX_APP.js
+const NX_DEFAULT_PROJECT = require('../workspace.json').defaultProject
+module.exports = NX_DEFAULT_PROJECT
+```
+
+```js
+// xdn/connector/build.js
+const path = require('path')
+const fs = require('fs-extra');
+const { DeploymentBuilder } = require('@xdn/core/deploy')
+const FrameworkBuildError = require('@xdn/core/errors/FrameworkBuildError')
+const NX_APP = require('../NX_APP')
+const rootDir = process.cwd()
+const distDir = path.join(rootDir, `./dist/apps/${NX_APP}`)
+const lambdaDir = path.join(rootDir, './.xdn/lambda')
+async function build({ skipFramework }) {
+  const builder = new DeploymentBuilder()
+  builder.clearPreviousBuildOutput()
+  if (!skipFramework) {
+    // run the NX build
+    try {
+      await builder.exec('nx build --prod')
+    } catch (e) {
+      // this lets the user know that the build error was within their application code, not their XDN router or configuration.
+      throw new FrameworkBuildError('NX')
+    }
+  }
+  // include nextjs server files
+  await fs.copy(path.join(distDir, '.next'), path.join(lambdaDir, '.next'))
+  
+  // build the XDN deployment bundle in the .xdn directory
+  await builder.build()
+}
+module.exports = build
+```
+
+```js
+// xdn/connector/prod.js
+const http = require('http')
+const url = require('url')
+const next = require('next')
+const dev = process.env.NODE_ENV !== 'production'
+const app = next({ dev })
+const handle = app.getRequestHandler()
```

Comment:
I believe @markbrocato had some concerns around the use of this connector and requiring of the full Next library. I will hold off on https://github.com/moovweb-docs/xdn-examples/pull/22 until further review.